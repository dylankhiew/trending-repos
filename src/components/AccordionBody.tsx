import { Image, StyleSheet, Text, View } from "react-native";

import { IMAGES } from "../../assets/IMAGES";
import { COLOR } from "../constants/colorConstants";
import { SPACING } from "../constants/spacingConstants";

interface AccordionBodyProps {
  repository: app.RepositoryItem;
}

export default function AccordionBody({
  repository,
}: AccordionBodyProps): JSX.Element {
  const { description, url, language, starsSince, forks, languageColor } =
    repository;
  const bodyDescription = `${description ?? ""} (${url})`;

  return (
    <View style={styles.accordionBodyContainer}>
      <Text style={styles.accordionDescription}>{bodyDescription}</Text>
      <View style={styles.accordionFooter}>
        <View style={styles.bottomFooter}>
          <View
            style={[styles.languageCircle, { backgroundColor: languageColor }]}
          />
          <Text style={styles.bottomFooterText}>{language}</Text>
        </View>
        <View style={styles.bottomFooter}>
          <Image source={IMAGES.ICON.STAR} style={styles.bottomFooterIcon} />
          <Text style={styles.bottomFooterText}>{starsSince}</Text>
        </View>
        <View style={styles.bottomFooter}>
          <Image source={IMAGES.ICON.FORK} style={styles.bottomFooterIcon} />
          <Text style={styles.bottomFooterText}>{forks}</Text>
        </View>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  accordionBodyContainer: {
    display: "flex",
    flexDirection: "column",
    flex: 1,
    margin: 0,
    // Offspace which accounts for avatar and its margin
    marginLeft: 32 + SPACING.S_4 + SPACING.S_4,
    marginBottom: SPACING.S_2,
  },
  accordionFooter: {
    flex: 1,
    flexDirection: "row",
    marginTop: SPACING.S_2,
  },
  bottomFooter: {
    flexDirection: "row",
    justifyContent: "center",
    alignContent: "center",
    alignItems: "center",
    marginRight: SPACING.S_3,
  },
  languageCircle: {
    height: 12,
    width: 12,
    borderRadius: 12,
    marginRight: SPACING.S_0,
  },
  bottomFooterIcon: {
    height: 16,
    width: 16,
  },
  bottomFooterText: {
    color: COLOR.GRAY300,
    fontFamily: "Roboto-Regular",
  },
  accordionDescription: {
    fontFamily: "PingFangSC-Regular",
    fontSize: 12,
    marginRight: SPACING.S_4,
    color: COLOR.GRAY300,
  },
});
