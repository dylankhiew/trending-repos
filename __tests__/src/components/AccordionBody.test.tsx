import React from "react";
import renderer from "react-test-renderer";

import AccordionBody from "../../../src/components/AccordionBody";

const repo: app.RepositoryItem = {
  rank: 1,
  username: "Nutlope",
  repositoryName: "roomGPT",
  url: "https://github.com/Nutlope/roomGPT",
  description:
    "Upload a photo of your room to generate your dream room with AI.",
  language: "TypeScript",
  languageColor: "#3178c6",
  totalStars: 2940,
  forks: 227,
  starsSince: 806,
  since: "daily",
  builtBy: [
    {
      username: "Nutlope",
      url: "https://github.com/Nutlope",
      avatar: "https://avatars.githubusercontent.com/u/63742054?s=40&v=4",
    },
  ],
};

it(" should render correctly", () => {
  const tree = renderer.create(<AccordionBody repository={repo} />).toJSON();
  expect(tree).toMatchSnapshot();
});
