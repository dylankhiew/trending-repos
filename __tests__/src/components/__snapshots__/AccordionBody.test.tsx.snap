// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` should render correctly 1`] = `
<View
  style={
    {
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
      "margin": 0,
      "marginBottom": 8,
      "marginLeft": 64,
    }
  }
>
  <Text
    style={
      {
        "color": "#52575C",
        "fontFamily": "PingFangSC-Regular",
        "fontSize": 12,
        "marginRight": 16,
      }
    }
  >
    Upload a photo of your room to generate your dream room with AI. (https://github.com/Nutlope/roomGPT)
  </Text>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "row",
        "marginTop": 8,
      }
    }
  >
    <View
      style={
        {
          "alignContent": "center",
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginRight": 12,
        }
      }
    >
      <View
        style={
          [
            {
              "borderRadius": 12,
              "height": 12,
              "marginRight": 4,
              "width": 12,
            },
            {
              "backgroundColor": "#3178c6",
            },
          ]
        }
      />
      <Text
        style={
          {
            "color": "#52575C",
            "fontFamily": "Roboto-Regular",
          }
        }
      >
        TypeScript
      </Text>
    </View>
    <View
      style={
        {
          "alignContent": "center",
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginRight": 12,
        }
      }
    >
      <Image
        source={1}
        style={
          {
            "height": 16,
            "width": 16,
          }
        }
      />
      <Text
        style={
          {
            "color": "#52575C",
            "fontFamily": "Roboto-Regular",
          }
        }
      >
        806
      </Text>
    </View>
    <View
      style={
        {
          "alignContent": "center",
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginRight": 12,
        }
      }
    >
      <Image
        source={1}
        style={
          {
            "height": 16,
            "width": 16,
          }
        }
      />
      <Text
        style={
          {
            "color": "#52575C",
            "fontFamily": "Roboto-Regular",
          }
        }
      >
        227
      </Text>
    </View>
  </View>
</View>
`;
